### PR TITLE
Beta development - Setting SDR Device ID, Fixed Receive scripts to pass PPM Offset, Fixed dependancies to work on OS Lite versions

### PR DESCRIPTION
**Pull request by creas002**
1) Created documentation for Setting SDR Device ID
2) Fixed receive_noaa.sh and receive_meteor.sh scripts to pass PPM Offset to satdump
3) Fixed dependencies.yml so it works correctly on both Full & Lite OS versions of Bookworm and Bullseye
...
OS Lite versions do not have .config directory already created and Full versions does.

When RN2 install executes on a Fresh OS Lite version, which is missing the .config directory it gets created as as part of the .config/meteordemod task in dependencies and its created incorrectly. 

The .config ownership of top directory is root:root 755 when this happens and it needs to be target_user:target_user 700

The result of .config ownership being wrong causes the verification tool which is called as part of running install_and_upgrade.sh to flag 4 items as failed which is its job.

richard@raspberrypi:~$ cat  /var/log/raspberry-noaa-v2/verification.log |grep FAIL
12-08-2024 17:23 FAIL : ****************************************************
12-08-2024 17:23 FAIL : CURRENT directory 755 root root /home/richard/.config
12-08-2024 17:23 FAIL : DESIRED directory 700 richard richard /home/richard/.config
12-08-2024 17:23 FAIL : ****************************************************
12-08-2024 17:23 FAIL : /home/richard/.config/satdump does not exist
12-08-2024 17:23 FAIL : /home/richard/.config/satdump/satdump_tles.txt does not exist
12-08-2024 17:23 FAIL : /home/richard/.config/satdump/settings.json does not exist

This problem does not occur on Fresh OS Full installations, but I still tested that the change did not introduce any issue on fresh OS Full installs.

### CODE REVIEW

**Review by MihajloPi — COMMENTED** [review]
This looks good, although I think it's safe to say we can delete "annotation directory" Ansible task since we don't have use for this directory.

**Review by MihajloPi — APPROVED** [review]
Looking good!

File: scripts/receive_meteor.sh [hunk 1]
```diff
@@ -169,7 +169,7 @@ polar_direction=0
 
 log "Recording ${NOAA_HOME} via $receiver at ${METEOR_FREQUENCY} MHz using SatDump record " "INFO"
 audio_temporary_storage_directory="$(dirname "${RAMFS_FILE_BASE}")"
-$SATDUMP live meteor_m2-x_lrpt${mode} "$audio_temporary_storage_directory" --source $receiver --samplerate $samplerate --frequency "${METEOR_FREQUENCY}e6" $sdr_id_option $SDR_DEVICE_ID $gain_option $GAIN $bias_tee_option $finish_processing --fill_missing --timeout $CAPTURE_TIME >> $NOAA_LOG 2>&1
+$SATDUMP live meteor_m2-x_lrpt${mode} "$audio_temporary_storage_directory" --source $receiver --samplerate $samplerate --ppm_correction ${FREQ_OFFSET} --frequency "${METEOR_FREQUENCY}e6" $sdr_id_option $SDR_DEVICE_ID $gain_option $GAIN $bias_tee_option $finish_processing --fill_missing --timeout $CAPTURE_TIME >> $NOAA_LOG 2>&1
```
Comment (jp112sdl) [inline scripts/receive_meteor.sh:172]: Afaik, `--ppm_correction` is only valid for RTL based receivers (RTL-SDR and RTL-TCP)

File: scripts/receive_noaa.sh [hunk 2]
```diff
@@ -167,7 +167,7 @@ daylight=$((SUN_ELEV > SUN_MIN_ELEV ? 1 : 0))
 #start capture
 log "Recording ${NOAA_HOME} via ${RECEIVER_TYPE} at ${freq} MHz via SatDump live pipeline" "INFO"
 audio_temporary_storage_directory="$(dirname "${RAMFS_FILE_BASE}")"
-$SATDUMP live noaa_apt $audio_temporary_storage_directory --source $receiver --samplerate $samplerate --frequency "${NOAA_FREQUENCY}e6" --satellite_number ${SAT_NUMBER} --sdrpp_noise_reduction $sdr_id_option $SDR_DEVICE_ID $gain_option $GAIN $bias_tee_option $crop_topbottom --start_timestamp $PASS_START --save_wav $finish_processing --timeout $CAPTURE_TIME >> $NOAA_LOG 2>&1
+$SATDUMP live noaa_apt $audio_temporary_storage_directory --source $receiver --samplerate $samplerate --ppm_correction ${FREQ_OFFSET} --frequency "${NOAA_FREQUENCY}e6" --satellite_number ${SAT_NUMBER} --sdrpp_noise_reduction $sdr_id_option $SDR_DEVICE_ID $gain_option $GAIN $bias_tee_option $crop_topbottom --start_timestamp $PASS_START --save_wav $finish_processing --timeout $CAPTURE_TIME >> $NOAA_LOG 2>&1
```
Comment (jp112sdl) [inline scripts/receive_noaa.sh:170]: Afaik, `--ppm_correction` is only valid for RTL based receivers (RTL-SDR and RTL-TCP)

Comment (creas002) [reply scripts/receive_noaa.sh:170]: That would explain why it was never fully implemented. Thanks Jerome. That explains why it was never fully implemented.  I will work get it put the original back and then working on conditionally passing value based on receiver type.  Do you know who started implementing PPM offset? It was 95% baked already and so close.

Comment (jp112sdl) [reply scripts/receive_noaa.sh:170]: Before reception was done by a single script we had different scripts for each type of SDR.
And the `rtl_fm` script had the `-p` parameter.
https://github.com/jekhokie/raspberry-noaa-v2/blob/e6bca5549a884151b4936895f161c855b8d67905/scripts/audio_processors/noaa_record_rtl_fm.sh

**Pull request merged**